db/low-level/src/ts/dns.ts

```diff
@@ -0,0 +1,241 @@
+        isVersioned: boolean;
+        createdOn: Date;
+        modifiedOn: Date;
+    };
+    subDirectories: string[];
+    files: string[];
+}
+
+export type DnsServiceList = string[];
+export type DnsLongNameList = string[];
+
+export class DnsClient extends ApiClient {
+    private mkendpoint(endpoint: string): string {
+        return this.endpoint + `/dns${endpoint}`;
+    }
+
+    constructor(conf: ApiClientConfig) {
+        super(conf);
+    }
+
+    /** @arg longName - public name that can be shared
+     *  @returns a promise to say if the longName was registered
+     */
+    //NOTE only 'human readable' longNames are accepted
+    // How that is defined, I have no clue, but try to use real words no random capitals
+    public async register(longName: string): Promise<void> {
+        const response = await saneResponse(WebRequest.post(
+            this.mkendpoint(`/${longName}`), {
+                auth: {
+                    bearer: (await this.authRes).token
+                }
+            }));
+
+        if (response.statusCode !== 200) {
+            throw new SafeError(`statusCode=${response.statusCode} !== 200`, response);
+        }
+    }
+
+    /** @arg longName - public name that can be shared
+     *  @arg serviceName - name of service mapped to longName
+     *  @arg rootPath - app or drive
+     *  @arg serviceHomeDirPath - the full path of the directory to be associated with to the service
+     *  @returns a promise to say if the longName and service were registered
+     */
+    public async registerAndAddService(longName: string, serviceName: string,
+                                       rootPath: RootPath, serviceHomeDirPath: string): Promise<void> {
+        let Body = {
+            "longName": longName,
+            "serviceName": serviceName,
+            "rootPath": rootPath,
+            "serviceHomeDirPath": serviceHomeDirPath
+        }
+
+        const response = await saneResponse(WebRequest.post(
+            this.mkendpoint(""), {
+                auth: {
+                    bearer: (await this.authRes).token
+                },
+                json: true,
+                body: Body
+            }));
+
+        if (response.statusCode !== 200) {
+            throw new SafeError(`statusCode=${response.statusCode} !== 200`, response);
+        }
+    }
+
+    /** @arg longName - public name that can be shared
+     *  @returns a list of services mapped to the longName
+     */
+    public async getServices(longName: string): Promise<DnsServiceList> {
+        const response = await saneResponse(WebRequest.get(
+            this.mkendpoint(`/${longName}`), {
+                auth: {
+                    bearer: (await this.authRes).token
+                }
+            }));
+
+        if (response.statusCode !== 200) {
+            throw new SafeError(`statusCode=${response.statusCode} !== 200`, response);
+        }
+
+        return JSON.parse(response.content); //little sketchy
```

Comment:
If you want you could make an isDnsServiceList predicate function which checks the type of the parsed result manually. An example of this is the `isStructredDataMetadata` predicate in the structured-data.ts file. That one only does a kind of half-assed type checking though.